src/ThebigwordTranslatorUi.php

```diff
@@ -0,0 +1,141 @@
+<?php
+
+/**
+ * @file
+ * Contains Drupal\tmgmt_thebigword\ThebigwordTranslatorUi.
+ */
+
+namespace Drupal\tmgmt_thebigword;
+
+use Drupal\Core\Datetime\DrupalDateTime;
+use Drupal\tmgmt\TranslatorPluginUiBase;
+use Drupal\Core\Form\FormStateInterface;
+use Drupal\tmgmt\JobInterface;
+
+/**
+ * Thebigword translator UI.
+ */
+class ThebigwordTranslatorUi extends TranslatorPluginUiBase {
+
+  /**
+   * {@inheritdoc}
+   */
+  public function buildConfigurationForm(array $form, FormStateInterface $form_state) {
+    $form = parent::buildConfigurationForm($form, $form_state);
+
+    /** @var \Drupal\tmgmt\TranslatorInterface $translator */
+    $translator = $form_state->getFormObject()->getEntity();
+
+    $form['client_contact_key'] = array(
+      '#type' => 'textfield',
+      '#title' => t('Thebigword client contact key'),
+      '#default_value' => $translator->getSetting('client_contact_key'),
+      '#description' => t('Please enter your client contact key.'),
+    );
+    $form += parent::addConnectButton();
+
+    return $form;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function validateConfigurationForm(array &$form, FormStateInterface $form_state) {
+    parent::validateConfigurationForm($form, $form_state);
+    /** @var \Drupal\tmgmt\TranslatorInterface $translator */
+    $translator = $form_state->getFormObject()->getEntity();
+    $supported_remote_languages = $translator->getPlugin()->getSupportedRemoteLanguages($translator);
```

Comment:
is there no better way to do this? doing an actual request (some kind of ping or generic information)  and checking if that works?